src/consumer/consumer.js

```diff
@@ -268,6 +271,59 @@ export default class Consumer {
     return bitId.changeVersion(version);
   }
 
+  /**
+   * throws a ComponentNotFound exception if not found in the model
+   */
+  async loadComponentFromModel(id: BitId): Promise<Component> {
+    const modelComponent: ModelComponent = await this.scope.getModelComponent(id);
+    if (!id.version) throw new TypeError('consumer.loadComponentFromModel, version is missing from the id');
+    return modelComponent.toConsumerComponent(id.version, this.scope.name, this.scope.objects, this.bitMap);
+  }
+
+  /**
+   * return a component only when it's stored locally.
+   * don't go to any remote server and don't throw an exception if the component is not there.
+   */
+  async loadComponentFromModelIfExist(id: BitId): Promise<?Component> {
+    return this.loadComponentFromModel(id).catch((err) => {
```

Comment:
@davidfirst why catching an error instead of add arg to throw/not throw to the loadComponentFromModel

Reply:
I was actually debating between these two options. The thing is that if you have one function with a parameter, the return value can be null (?Component) and then, every time you use this function you have to check whether you got a null value, otherwise, Flow makes you trouble.
Also, in my option, it looks cleaner this way. Less parameters in the methods and the code is easier to read.